assets/js/ajax-filters.js

```diff
@@ -1,288 +1,458 @@
+		var $results = $target.find( '.job_listings' );
+		var $showing = $target.find( '.showing_jobs' );
 
-	$( document.body ).on( 'click', '.load_more_jobs', function() {
-		var target           = $( this ).closest( 'div.job_listings' );
-		var page             = parseInt( ( $( this ).data( 'page' ) || 1 ), 10 );
-		var loading_previous = false;
+		if ( typeof append !== 'boolean' ) {
+			append = false;
+		}
 
-		$(this).addClass( 'loading' );
+		if ( typeof result.showing === 'string' && result.showing ) {
+			var $showing_el = jQuery( '<span>' ).html( result.showing );
+			$showing
+				.show()
+				.html( '' )
+				.html( result.showing_links )
+				.prepend( $showing_el );
+		} else {
+			$showing.hide();
+		}
 
-		if ( $(this).is('.load_previous') ) {
-			page             = page - 1;
-			loading_previous = true;
-			if ( page === 1 ) {
-				$(this).remove();
+		if ( result.showing_all ) {
+			$showing.addClass( 'wp-job-manager-showing-all' );
+		} else {
+			$showing.removeClass( 'wp-job-manager-showing-all' );
+		}
+
+		if ( result.html ) {
+			if ( append ) {
+				$results.append( result.html );
 			} else {
-				$( this ).data( 'page', page );
+				$results.html( result.html );
+			}
+		}
+
+		if ( true === $target.data( 'show_pagination' ) ) {
+			$target.find( '.job-manager-pagination' ).remove();
+
+			if ( result.pagination ) {
+				$target.append( result.pagination );
 			}
 		} else {
-			page = page + 1;
-			$( this ).data( 'page', page );
-			job_manager_store_state( target, page );
+			if ( ! result.found_jobs || result.max_num_pages <= result.data.page ) {
+				$( '.load_more_jobs:not(.load_previous)', $target ).hide();
+			} else {
+				$( '.load_more_jobs', $target ).show();
+			}
+			$( '.load_more_jobs', $target )
+				.removeClass( 'loading' )
+				.data( 'page', result.data.page );
+			$( 'li.job_listing', $results ).css( 'visibility', 'visible' );
 		}
 
-		target.triggerHandler( 'update_results', [ page, true, loading_previous ] );
-		return false;
-	} );
+		return true;
+	}
 
-	$( 'div.job_listings' ).on( 'click', '.job-manager-pagination a', function() {
-		var target = $( this ).closest( 'div.job_listings' );
-		var page   = $( this ).data( 'page' );
+	var xhr = [];
+	$( 'div.job_listings' )
+		.on( 'click', 'li.job_listing a', function() {
+			var $target = $( this ).closest( 'div.job_listings' );
+
+			// We're moving away to a job listing. Let's make sure the results persist.
+			job_manager_persist_results( $target, true );
+		} )
+		.on( 'click', '.job-manager-pagination a', function() {
+			var $target = $( this ).closest( 'div.job_listings' );
+			var page = $( this ).data( 'page' );
+
+			$target.triggerHandler( 'update_results', [ page, false ] );
+
+			$( 'body, html' ).animate(
+				{
+					scrollTop: $target.offset().top,
+				},
+				600
+			);
+
+			return false;
+		} )
+		.on( 'update_results', function( event, page, append ) {
+			var data = '';
+			var $target = $( this );
```

Comment:
Are we using the same prettier configuration as Calypso? Seems like it should still align assignments for us, but maybe not. It's been a while since I've looked at Calypso code 🙂

Reply:
Yes, I stole the config file from Calypso [[#](https://github.com/Automattic/wp-calypso/blob/master/client/blocks/site-preview/index.jsx#L25-L32)]. We could use our own, though, if we'd prefer.

Reply:
Ok, I was just surprised at the lack of alignment. But if it's consistent with Calypso then it's fine by me 👍